src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -1184,6 +1188,16 @@ public long getStartItemCount() {
         return this.startItemCount;
     }
 
+    private TileEntity getTile(ICraftingMedium craftingProvider) {
+        try {
+            Method method = craftingProvider.getClass().getDeclaredMethod("getTile");
```

Comment:
I didn't mean to use reflection to find a method by name, you can add a default implementation to the interface like that:
```java
default TileEntity getTile() {return null;}
```
That way you can add the extra method to the interface, but still be backwards compatible

Reply:
> I didn't mean to use reflection to find a method by name, you can add a default implementation to the interface like that:
> 
> ```java
> default TileEntity getTile() {return null;}
> ```
> 
> That way you can add the extra method to the interface, but still be backwards compatible

you can't just set by defalut

Reply:
In an interface you can: <https://www.baeldung.com/java-static-default-methods>

Reply:
![image](https://user-images.githubusercontent.com/33802162/216993488-11f8b1d6-38c6-4d63-9859-2283694a29a9.png)

Reply:
![image](https://user-images.githubusercontent.com/33802162/216996271-582fc206-316a-47cb-8fbd-d44d4919104b.png)

Reply:
If you rename it to something like `getHostTile()` it works. (do you even need the TileEntity? maybe a DimensionalCoord would be enough? `default DimensionalCoord getHostPosition() {return null;}` for example)

Reply:
it's absolutely need getTile()

Reply:
i don't want to rewrite using `ICraftingProvider` entity  for example ae2fc autofiller

Reply:
Ok in this case put a comment explaining all this above the reflection